Description: Change to parse the new format of named.stats.  The bug is filed
 upstream, but has not been closed
Origin: other, http://munin.projects.linpro.no/ticket/627
Bug: http://munin.projects.linpro.no/ticket/627
Index: munin-1.2.6/node/node.d/bind9_rndc.in
===================================================================
--- munin-1.2.6.orig/node/node.d/bind9_rndc.in	2009-05-28 20:17:39.000000000 +0200
+++ munin-1.2.6/node/node.d/bind9_rndc.in	2009-05-28 20:19:02.000000000 +0200
@@ -41,6 +41,7 @@
 open(my $stats, '<', $querystats) or die "$querystats: $!\n";
 seek($stats , 400, -1);		# go nearly to the end of the file
 				# to avoid reading it all
+my $mode = 0;                   # 0=Bind9.3, 1=Bind9.5
 
 while (my $line = <$stats>) {
     chomp $line;
@@ -58,6 +59,17 @@
     if ($line =~ m/\+\+\+ Statistics Dump \+\+\+/) {
 	# reset
 	undef %IN;
+    } elsif ($line =~ m/\+\+ Name Server Statistics \+\+/) {
+       undef %IN;
+       $mode = 1;
+
+    } elsif( $mode == 1 ) {
+       my($i,$s) = $line =~ m/^\s+(\d+) (.*)/  or next;
+       $s =~ /(successful answer|(?:non )?authoritative answer|recursion)/  or next;
+       my $key = $1;
+       $key =~ s/ /_/g;
+       $IN{$key} = $i;
+
     } else {
 	my ($what, $nb)= split('\s+', $line);
 	if ($what && ($what ne '---')) {
